### PR TITLE
[RSDK-5248] make the MPU6050 talk to the I2C bus directly

### PR DESCRIPTION
**Pull request by penguinland**
...instead of going through the board component to do it.

To get the tests to work, I've split `NewMpu6050` into that function (which constructs a real I2C connection) and `makeMpu6050` (which takes in an I2C connection plus all the same things as `NewMpu6050` and constructs everything else). That latter command can be called during tests to inject a mock I2C bus.

When folks are happy with this code, before I merge this PR I should write a migration script to update the configs in production.

I'm tagging @gvaradarajan for a review because he wrote most of the tests here.

### CODE REVIEW

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 1]
```diff
@@ -17,43 +17,27 @@ import (
...
-		}
+		cfg := Config{}
 		deps, err := cfg.Validate("path")
 		expectedErr := utils.NewConfigValidationFieldRequiredError("path", "i2c_bus")
 		test.That(t, err, test.ShouldBeError, expectedErr)
 		test.That(t, deps, test.ShouldBeEmpty)
 	})
 
-	t.Run("adds board name to dependencies on success", func(t *testing.T) {
+	t.Run("no dependencies on success", func(t *testing.T) {
```
Comment (gvaradarajan) [inline components/movementsensor/mpu6050/mpu6050_test.go:28]: [q] do we even want this test? Seems like it's not testing anything important anymore. If the board dependency is never used, why does it matter whether it is in the dependencies or not?

**Review by gvaradarajan — APPROVED** [review]
LGTM, just one question

File: components/movementsensor/mpu6050/mpu6050_test.go (hunk 1 above)
Comment (penguinland) [reply components/movementsensor/mpu6050/mpu6050_test.go]: Good point, I'll remove it!

**Comment by penguinland** [comment]
...and the config migration script is merged. Let's go!

**Pull request merged**